pkgs/development/python-modules/flax/default.nix

```diff
@@ -37,24 +40,27 @@ buildPythonPackage rec {
   ];
 
   propagatedBuildInputs = [
-    cloudpickle
     jax
-    matplotlib
     msgpack
     numpy
     optax
+    pyyaml
     rich
     tensorstore
+    typing-extensions
```

Comment:
why add pyyaml and typing-extensions if flax built fine without them before this PR?

Reply:
I added them because they were mentioned in the `pryproject.toml`.
They might not be necessary for the test suite to complete without error but they might be required for certain usage of `flax`.

Actually, I don't know what is the recommended approach for packaging python libraries in nixpkgs:
- Trusting the upstream dependency list and including as many of the said-to-be required packages to the derivation;
- or having the `propagatedBuildInputs` be as limited as possible (so that the package builds fine and the tests succeed).

@K900 @dotlambda @SuperSandro2000 what do you think about that ?

Reply:
IMO: we should include all dependencies instead of relying on the indirectly brought in ones.

Reply:
Makes sense, transitive dependency propagation bites again...